2018/readme.md

```diff
@@ -0,0 +1,128 @@
+
+`rules_engine` in [`rules_engine.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/rules_engine.py) 
+
+Contains logic for each edit. The rules engine can generate an edit report for a passed file. This report contains the following fields:
+
+    * edit name
+    * status (pass/fail)
+    * number of rows failed
+    * ULIs/NULIs of rows that failed (as a list)
+    
+The report is created as a dataframe that can be passed to other methods of the class and is also written to CSV.
+
+LAR and TS dataframes are passed into this class, and its methods output edit details that show among its fields the edit name, the status of the edit (failed or passed), the number of rows failed by the edit, if any, and the ULI's or NULIs (loan ID) of the rows that fail the edit. Each method creates a results dataframe that holds the rows that have failed the specific edit. The class then wraps the results into an edit report that can be parsed by other methods in the repository.  
+
+`lar_constraints` in [`lar_constraints.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/rules_engine.py) 
+
+This class contains methods that modify synthetic LAR rows that fail edits so that they pass those edits. Because the lar_constraints class changes LAR rows sequentially and with set values, it may change values to pass an edit, while failing edits previously corrected. To create clean files, the synthetic rows are passed between lar_constraints and rules_engine, until there are no more syntax or validity edits present. 
+
+`test_data` in [`test_file_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/test_file_generator.py) 
+
+Contains methods that manipulate LAR or TS dataframes to fail a specific syntax or validity edit. Because the test_data class changes LAR rows sequentially and with set values, it may fail other edits while changing values in the synthetic LAR to fail a specific edit.
+
+Note: changes to the clean data do not create exhaustive test cases for each edit. 
+
+Each edit logic method in `lar_constraints`, `test_data`, and `rules_engine` is prefaced with 's', 'v', or 'q' to indicate a syntax, validity or quality edit respectively.
+
+## Configurations 
+The following yaml configuration files and values can be changed to alter the output of HMDA submission test files. 
+
+### [Test Files Configuration](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/test_filepaths.yaml)
+
+| Key| Default Value | Expanation  |
+|:-------------|:-------------|:--------|
+|clean_filepath | "../edits_files/clean_files/{bank_name}/" | Filepath for generated clean files. Note: the filepath needs to contain `{bank_name}` for formatting the institution name to the path. |
```

Comment:
Can you change the quotes around default values to ` characters so that they render like a code value?

Reply:
Sure - I can add this.